examples/memnn/memnn.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
+
+import argparse
+import collections
+import copy
+import glob
+import sys
+
+import numpy
+import six
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+from chainer import initializers
+import chainer.links as L
+from chainer import optimizers
+from chainer import training
+from chainer.training import extensions
+
+import babi
+
+
+class BoWEncoder(object):
```

Comment:
Isn't it simpler to make it a function instead of a callable class? (If you intended to emphasize that encoder, in general, may have its state (though the `BoWEncoder` does not), it's ok to leave it as is)

Reply:
I thought when I use `self.f = f` and `f` is a function, `self.f(x)` calls `f(self, x)`. I now test it and it works. I'll use just a function.